File: src/examples/javascript/webapp/web_server.js
load('vertx.js');

var server = new vertx.HttpServer()
    .setSSL(true)
    .setKeyStorePath('server-keystore.jks')
    .setKeyStorePassword('wibble');

// Serve the static resources
server.requestHandler(function(req) {
  if (req.path === '/') {
    req.response.sendFile('web/index.html');
  } else if (req.path.indexOf('..') === -1) {
    req.response.sendFile('web' + req.path);
  } else {
    req.response.statusCode = 404;
    req.response.end();
  }
})

new vertx.SockJSBridge(server, {prefix : '/eventbus'},
  [
    // Allow calls to get static album data from the persistor
    {
      address : 'demo.persistor',
      match : {
        action : 'find',
        collection : 'albums'
      }
    },
    // Allow user to login
    {
      address : 'demo.authMgr.login'
    },
    // Let through orders posted to the order manager
    {
      address : 'demo.orderMgr'
    }
  ]
);

server.listen(8080, 'localhost');

function vertxStop() {
  server.close();
}

File: src/examples/ruby/webapp/web_server.rb
require('vertx')
include Vertx

@server = HttpServer.new
@server.ssl = true
@server.key_store_path = 'server-keystore.jks'
@server.key_store_password = 'wibble'

# Serve the static resources
@server.request_handler do |req|
  if req.path == '/'
    req.response.send_file('web/index.html')
  elsif !req.path.include?('..')
    req.response.send_file('web' + req.path)
  else
    req.response.status_code = 404
    req.response.end
  end
end

# Link up the client side to the server side event bus
Vertx::SockJSBridge.new(@server, {'prefix' => '/eventbus'},
 [
    # Allow calls to get static album data from the persistor
    {
      'address' => 'demo.persistor',
      'match' => {
        'action' => 'find',
        'collection' => 'albums'
      }
    },
    # Allow user to login
    {
      'address' => 'demo.authMgr.login'
    },
    # Let through orders posted to the order manager
    {
      'address' => 'demo.orderMgr'
    }
  ])

@server.listen(8080, 'localhost')

def vertx_stop 
  server.close
end
